src/main/java/org/opentripplanner/routing/edgetype/rentedgetype/RentVehicleEdge.java

```diff
@@ -0,0 +1,83 @@
+package org.opentripplanner.routing.edgetype.rentedgetype;
+
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.StateEditor;
+import org.opentripplanner.routing.core.vehicle_sharing.VehicleDescription;
+import org.opentripplanner.routing.edgetype.TemporaryEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.vertextype.TemporaryRentVehicleVertex;
+
+import java.util.Locale;
+
+/**
+ * This edge allows us to rent vehicle (or leave current vehicle and rent new one).
+ * This edge is a loop on {@link TemporaryRentVehicleVertex} which, when traversed, changes our current traverse mode,
+ * but leaves us in the same location.
+ */
+public class RentVehicleEdge extends Edge implements TemporaryEdge {
+
+    private final VehicleDescription vehicle;
+
+    private final ParkingZoneInfo parkingZones;
+
+    private final ParkingZoneInfo parkingZonesEnabled;
+
```

Comment:
Trochę mylące jest to, że są dwie instancje tego. Adam twierdzi, że to są serwisy więc wgl jest to dziwne.

Reply:
To są serwisy i to w dodatku stateful, bo strefy parkowania mogą się zmieniać w czasie działania programu. Można rozważyć przeniesienie ich generowania do fazy budowy grafu, wtedy zniknie problem z updatowaniem i będzie można zrobić jedną klasę, która odpowiada za wyliczanie stref parkowania 👍 .